File: GoaOnline/src/Application.java
public class Application extends User {
    int applicationId;
    private int serviceId;
    private String applicationDate;
    private String status;
    private String[] documents;

    // Constructor
    public Application(int userId, String name, String email, String phoneNumber, String address, String userType,
                        int applicationId, int serviceId, String applicationDate, String status, String[] documents) {
        super(userId, name, email, phoneNumber, address, userType);
        this.applicationId = applicationId;
        this.serviceId = serviceId;
        this.applicationDate = applicationDate;
        this.status = status;
        this.documents = documents;
    }

    // Overloaded method
    public void submitApplication() {
        System.out.println("Application submitted.");
    }

    // Overloaded method with parameters
    public void submitApplication(int serviceId) {
        System.out.println("Application for service ID " + serviceId + " submitted.");
    }

    public void checkStatus() {
        System.out.println("Checking application status: " + status);
    }

    public void uploadDocuments(String[] docs) {
        this.documents = docs;
        System.out.println("Documents uploaded.");
    }

    public void editApplication() {
        System.out.println("Application edited.");
    }

    public int getApplicationId() {
        // TODO Auto-generated method stub
        return this.applicationId;
    }
}

File: python/application.py
from user import User

class Application(User):
    def __init__(self, user_id, name, email, phone_number, address, user_type,
                 application_id, service_id, application_date, status, documents):
        super().__init__(user_id, name, email, phone_number, address, user_type)
        self.application_id = application_id
        self.service_id = service_id
        self.application_date = application_date
        self.status = status
        self.documents = documents

    def submit_application(self, service_id=None):
        if service_id:
            print(f"Application for service ID {service_id} submitted.")
        else:
            print("Application submitted.")

    def check_status(self):
        print(f"Checking application status: {self.status}")

    def upload_documents(self, docs):
        self.documents = docs
        print("Documents uploaded.")

    def edit_application(self):
        print("Application edited.")

